packages/plugin-core/src/components/views/LookupV3QuickPickView.ts

```diff
@@ -0,0 +1,303 @@
+  LookupSelectionTypeEnum,
+} from "../lookup/ButtonTypes";
+import {
+  ILookupViewModel,
+  NameModifierMode,
+  SelectionMode,
+} from "../lookup/LookupViewModel";
+import { DendronQuickPickerV2, VaultSelectionMode } from "../lookup/types";
+
+/**
+ * A 'view' that represents the UI state of the Lookup Quick Pick. This
+ * essentially controls the button state of the quick pick and reacts upon user
+ * mouse clicks to the buttons.
+ */
+export class LookupV3QuickPickView implements Disposable {
+  private _quickPick: DendronQuickPickerV2;
+  private _viewState: ILookupViewModel;
+  private _disposables: Disposable[];
+  private _providerId?: string;
+
+  constructor(
+    quickPick: DendronQuickPickerV2,
+    viewModel: ILookupViewModel,
+    providerId?: string // For telemetry purposes only
+  ) {
+    this._quickPick = quickPick;
+    this._viewState = viewModel;
+    this._providerId = providerId;
+    this._disposables = [];
+
+    this.setupViewModel();
+
+    this._disposables.push(
+      this._quickPick.onDidTriggerButton(this.onTriggerButton)
+    );
+  }
+  dispose() {
+    this._disposables.forEach((callback) => callback.dispose());
+  }
+
+  private setupViewModel(): void {
+    const ToLinkBtn = this.getButton("selection2link");
+    const ExtractBtn = this.getButton("selectionExtract");
+    const ToItemsBtn = this.getButton("selection2Items");
+
+    this._disposables.push(
+      this._viewState.selectionState.bind(async (newValue) => {
+        switch (newValue) {
+          case SelectionMode.selection2Items: {
+            if (ToLinkBtn) ToLinkBtn.pressed = false;
+            if (ExtractBtn) ExtractBtn.pressed = false;
+            if (ToItemsBtn) ToItemsBtn.pressed = true;
+            break;
+          }
+
+          case SelectionMode.selection2Link: {
+            if (ToLinkBtn) ToLinkBtn.pressed = true;
+            if (ExtractBtn) ExtractBtn.pressed = false;
+            if (ToItemsBtn) ToItemsBtn.pressed = false;
+            break;
+          }
+          case SelectionMode.selectionExtract: {
+            if (ToLinkBtn) ToLinkBtn.pressed = false;
+            if (ExtractBtn) ExtractBtn.pressed = true;
+            if (ToItemsBtn) ToItemsBtn.pressed = false;
+            break;
+          }
+          case SelectionMode.None: {
+            if (ToLinkBtn) ToLinkBtn.pressed = false;
+            if (ExtractBtn) ExtractBtn.pressed = false;
+            if (ToItemsBtn) ToItemsBtn.pressed = false;
+            break;
+          }
+          default:
+            assertUnreachable(newValue);
+        }
+
+        const buttons: DendronBtn[] = [];
+
+        if (ToLinkBtn) buttons.push(ToLinkBtn);
+        if (ExtractBtn) buttons.push(ExtractBtn);
+        if (ToItemsBtn) buttons.push(ToItemsBtn);
+
+        this.updateButtonsOnQuickPick(...buttons);
+      })
+    );
+
+    // Vault Selection is mapped to 'other' for some reason; Fix Later.
+    const vaultSelectionBtn = this.getButton("other");
```

Comment:
Can't really recall why we left it at "other", but I don't think we have a reason not to give it a better name at this point haha.

Reply:
Updated to "selectVault"